pkg/doc/jose/decrypter.go

```diff
@@ -107,6 +132,52 @@ func buildEncryptedData(encAlg string, jwe *JSONWebEncryption) ([]byte, error) {
 	return json.Marshal(encData)
 }
 
+// extractRecipientHeaders will extract RecipientHeaders from headers argument
+func extractRecipientHeaders(headers map[string]interface{}) (*RecipientHeaders, error) {
+	// Since headers is a generic map, epk value is converted to a generic map by Serialize(), ie we lose RawMessage
+	// type of epk. We need to convert epk value (generic map) to marshaled json so we can call RawMessage.Unmarshal()
+	// to get the original epk value (RawMessage type).
+	mapData, ok := headers[HeaderEPK].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("JSON value is not a map (%#v)", headers[HeaderEPK])
+	}
+
+	epkBytes, err := json.Marshal(mapData)
+	if err != nil {
+		return nil, err
+	}
+
+	epk := json.RawMessage{}
+
+	err = epk.UnmarshalJSON(epkBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	alg := ""
```

Comment:
why are we allowing `alg` and `kid` to be empty?

Reply:
we don't, but if the sender didn't specify them (the sender being not an aries agent), then the default should be empty string, not `"<nil>"`

Reply:
that is a strange decision - aries-framework-go is for agents, not a generic JOSE library.

Reply:
JWE is not just for aries agents, it's generic.. It's implied that aries agents will always set these headers for a recipient, these conditions are in case someone (being a non aries agent) is sending a JWE message without these headers.